### PR TITLE
Use tqdm as a requirement, remove vendored tqdm module.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR replaces the vendored tqdm module with a tqdm requirement.

## Motivation and Context
The vendored tqdm module was outdated and it is easier to maintain a requirement. The vendored module was causing DeprecationWarnings: https://github.com/glotzerlab/signac-flow/issues/244#issuecomment-585280247

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [x] Breaking change<sup>1</sup>

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,3 +2,4 @@ signac>=1.0.0
 jinja2>=2.8
 cloudpickle
 deprecation>=2
+tqdm>=4.10.0
```
Comment (csadorf) [inline requirements.txt]:
```suggestion
tqdm~=4.10.0
```
[The tqdm package follows semantic versioning](https://tqdm.github.io/contributing/#semantic-versioning). Should we therefore maybe use the [compatible release](https://www.python.org/dev/peps/pep-0440/#compatible-release) specifier instead?

File: setup.py [hunk 2]
```diff
@@ -14,6 +14,8 @@
     'cloudpickle',
     # Deprecation management
     'deprecation>=2',
+    # Progress bars
+    'tqdm>=4.10.0',
```
Comment (csadorf) [inline setup.py]:
```suggestion
    'tqdm~=4.10.0',
```
See comment for `requirements.txt` above.

File: requirements.txt (hunk 1 above)
Comment (bdice) [reply requirements.txt]: @csadorf Our use of tqdm is essentially the simplest use case. It looks like v5 is in progress and I don't think they're going to break `from tqdm import tqdm` according to the list of breaking changes. https://github.com/tqdm/tqdm/milestone/8

Usually I would say compatible versions are preferable, but I don't think we need to worry about that here. If you feel strongly about that, feel free to make the change here and in the signac repo (that PR was already merged).

Comment (csadorf) [reply requirements.txt]: Fine, no need for changes.

**Comment by csadorf** [comment]
@yuanzhou0827 I hope you don't mind, but I figured this PR was so trivial that I merged it without your review.

**Comment by yuanzhou0827** [comment]
@csadorf Never mind! I jam sorry... I just saw the review request. Thank you!

**Pull request merged**